clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/consumer/HonoKafkaConsumer.java

```diff
@@ -115,7 +111,11 @@
      */
     protected final LifecycleStatus lifecycleStatus = new LifecycleStatus();
 
-    private final AtomicReference<Pair<Promise<Void>, Promise<Void>>> subscriptionUpdatedAndPartitionsAssignedPromiseRef = new AtomicReference<>();
+    private final AtomicReference<Promise<Void>> initialPartitionAssignmentDonePromiseRef = new AtomicReference<>();
+
+    private final AtomicBoolean subscriptionUpdateTriggered = new AtomicBoolean();
+    private final Map<String, Promise<Void>> subscriptionUpdateTrackersForToBeAddedTopics = new ConcurrentHashMap<>();
```

Comment:
I guess the key is the name of the topic to be added?

Reply:
Yes, I've added a javadoc description here.